src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java

```diff
@@ -0,0 +1,217 @@
+    private final List<JSONObject> resultMeta;
+    private final JSONObject resultAggregation;
+    private final ProjectType projectType;
+    private final PrintStream logger;
+    private Build browserStackBuild;
+
+    public BrowserStackReportForBuild(final Run<?, ?> build, final ProjectType projectType, final String buildName, final PrintStream logger) {
+        super();
+        setBuild(build);
+        this.buildName = buildName;
+        this.browserStackSessions = new ArrayList<>();
+        this.result = new ArrayList<>();
+        this.resultMeta = new ArrayList<>();
+        this.resultAggregation = new JSONObject();
+        this.projectType = projectType;
+        this.logger = logger;
+        fetchBuildAndSessions();
+    }
+
+    private void fetchBuildAndSessions() {
+        final BrowserStackBuildAction browserStackBuildAction = getBuild().getAction(BrowserStackBuildAction.class);
+        if (browserStackBuildAction == null) {
+            log(logger, "Error: No BrowserStackBuildAction found");
+            return;
+        }
+
+        final BrowserStackCredentials credentials = browserStackBuildAction.getBrowserStackCredentials();
+        if (credentials == null) {
+            log(logger, "Error: BrowserStack credentials could not be fetched");
+            return;
+        }
+
+        BrowserStackClient client;
+        if (projectType == ProjectType.APP_AUTOMATE) {
+            client = new AppAutomateClient(credentials.getUsername(), credentials.getDecryptedAccesskey());
+        } else {
+            client = new AutomateClient(credentials.getUsername(), credentials.getDecryptedAccesskey());
+        }
+
+        this.browserStackBuild = fetchBrowserStackBuild(client, this.buildName);
+        if (this.browserStackBuild != null) {
+            this.browserStackSessions.addAll(fetchBrowserStackSessions(client, this.browserStackBuild.getId()));
+        }
+    }
+
+    public boolean generateBrowserStackReport() {
+        if (this.result.size() == 0) {
+            this.result.addAll(generateSessionsCollection(this.browserStackSessions));
+
+            if (this.result.size() > 0) {
+                this.result.sort(new SessionsSortingComparator());
+                this.resultMeta.addAll(this.result.subList(0, Math.min(this.result.size(), RESULT_META_MAX_SIZE)));
+                generateAggregationInfo();
+                return true;
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private Build fetchBrowserStackBuild(@Nonnull BrowserStackClient client, @Nonnull String buildName) {
+        Build build = null;
+        try {
+            build = client.getBuildByName(buildName);
+        } catch (BuildNotFound bnfException) {
+            log(this.logger, "No build found by name: " + buildName);
+        } catch (BrowserStackException bstackException) {
+            log(this.logger, "BrowserStackException occurred while fetching build: " + bstackException.toString());
```

Comment:
remove`this` wherever used while accessing

Reply:
done